### DISCUSSION TITLE
Enhancement ideas for saved/featured specimen capability

### DISCUSSION THREAD

**Discussion by liammulh** [numberscope/frontscope#190] [Ideas]
Eventually, we want presets in Numberscope.

**Comment by liammulh** [comment]
From 2022-11-10 student meeting:

- message or description along with presets
- similar to the demo, just a description on the side
- getPresets func
  - return collection of param objects, and string of description of the preset
  - assemble the presets
- submit preset as good
  - presets are handled higher level, separately
- visualizer decides where to put text?
- store preset in database
  - as JSON string?
- we need to be able to preserve state
  - we should use a consistent scheme for persisting state
- it would be nice if state (params?) could live in URL
  - then easily shareable
  - use client-side routing?

**Comment by gwhitney** [comment]
This is all implemented in ui2, except for the idea of allowing longer-form text descriptions of presets. So renaming the discussion to reflect that.
